apps/src/dance/ai/AiExplanationView.tsx

```diff
@@ -0,0 +1,162 @@
+import React, {useState} from 'react';
+const ToggleGroup = require('@cdo/apps/templates/ToggleGroup').default;
+import color from '@cdo/apps/util/color';
+import {CachedWeightsMapping} from './DanceAiClient';
+
+import CachedPalettes from '@cdo/static/dance/ai/model/cached-spacy-palette-map.json';
+import CachedBackgrounds from '@cdo/static/dance/ai/model/cached-spacy-background-map.json';
+import CachedForegrounds from '@cdo/static/dance/ai/model/cached-spacy-foreground-map.json';
+
+import {
+  Chart as ChartJS,
+  ChartOptions,
+  CategoryScale,
+  LinearScale,
+  BarElement,
+  Title,
+  Tooltip,
+  Legend,
+} from 'chart.js';
+import {Bar} from 'react-chartjs-2';
+
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  BarElement,
+  Title,
+  Tooltip,
+  Legend
+);
+
+enum FieldKey {
+  BACKGROUND_EFFECT = 'backgroundEffect',
+  FOREGROUND_EFFECT = 'foregroundEffect',
+  BACKGROUND_PALETTE = 'backgroundColor',
+}
+
+interface Result {
+  [FieldKey.BACKGROUND_EFFECT]: string;
+  [FieldKey.FOREGROUND_EFFECT]: string;
+  [FieldKey.BACKGROUND_PALETTE]: string;
+}
+
+interface AiExplanationViewProps {
+  inputs: string[];
+  result: Result;
+}
+
+interface FieldObject {
+  name: string;
+  data: CachedWeightsMapping;
+}
+
+interface Fields {
+  [FieldKey.BACKGROUND_EFFECT]: FieldObject;
+  [FieldKey.FOREGROUND_EFFECT]: FieldObject;
+  [FieldKey.BACKGROUND_PALETTE]: FieldObject;
+}
+
+/**
+ * Previews the blocks generated by the AI block in Dance Party.
+ */
+const AiExplanationView: React.FunctionComponent<AiExplanationViewProps> = ({
+  inputs,
+  result,
+}) => {
+  const [currentFieldKey, setCurrentFieldKey] = useState(
+    FieldKey.BACKGROUND_EFFECT
+  );
+
+  const fields: Fields = {
+    [FieldKey.BACKGROUND_EFFECT]: {
+      name: 'Background effect',
+      data: CachedBackgrounds as unknown as CachedWeightsMapping,
+    },
+    [FieldKey.FOREGROUND_EFFECT]: {
+      name: 'Foreground effect',
+      data: CachedForegrounds as unknown as CachedWeightsMapping,
+    },
+    [FieldKey.BACKGROUND_PALETTE]: {
+      name: 'Background palette',
+      data: CachedPalettes as unknown as CachedWeightsMapping,
+    },
+  };
+
+  const currentField = fields[currentFieldKey];
+
+  const options: ChartOptions<'bar'> = {
+    scales: {
+      y: {
+        stacked: true,
+      },
+      x: {
+        stacked: true,
+        ticks: {
+          color: function (context) {
+            if (context.tick.label === result[currentFieldKey]) {
+              return 'rgba(54, 162, 235, 1)';
+            }
+            return '#000000';
+          },
+        },
+      },
+    },
+  };
+
+  const colors = [
+    'rgb(255, 99, 132)',
+    'rgb(75, 192, 192)',
+    'rgb(53, 162, 235)',
```

Comment:
These feel like a magic number. Can we grab these from a constants file instead?

Reply:
All of these values are temporary and will be fixed in the styling pass.